Source/WebKit/NetworkProcess/webrtc/NetworkMDNSRegister.cpp

```
@@ -47,6 +47,14 @@ namespace WebKit {
 NetworkMDNSRegister::NetworkMDNSRegister(NetworkConnectionToWebProcess& connection)
     : m_connection(connection)
 {
+#if USE(GLIB)
+    GUniqueOutPtr<GError> error;
+    m_dbusProxy = adoptGRef(g_dbus_proxy_new_for_bus_sync(G_BUS_TYPE_SYSTEM,
+        static_cast<GDBusProxyFlags>(G_DBUS_PROXY_FLAGS_DO_NOT_CONNECT_SIGNALS | G_DBUS_PROXY_FLAGS_DO_NOT_LOAD_PROPERTIES), nullptr,
+        "org.freedesktop.Avahi", "/", "org.freedesktop.Avahi.Server", nullptr, &error.outPtr()));
+    if (error)
+        LOG_ERROR("Unable to connect to the Avahi daemon: %s", error->message);
```

A bit harsh? I think it's expected for avahi to not be running. I would either ignore the error here, or better: figure out which error is returned if avahi is not running, and print an error only if it's not that.

I'm not familiar with `LOG_ERROR()`; what exactly does that do? Is that release logging straight to the journal? I think we normally use `g_warning()` for stuff like this to ensure it appears in stderr.

---

Thanks for the review, I'll try to improve the patch next week :)

> I'm not familiar with LOG_ERROR(); what exactly does that do?

It expands to a `WTFReportError()` call which prints the msg to stderr.